### PR TITLE
Add `remove array element` in forms

### PR DESCRIPTION
**Pull request by goreck888**
In the backoffice service form added remove link for array fields: multimedia, use_cases_url, changelog, related_platforms etc.

### CODE REVIEW

**Comment by agpul** [comment]
Can I test this PR? How? Can you add a description?

**Comment by bwilk** [comment]
@agpul this is about removing array items. Looks ugly but works. Maybe the frontend logic could be better if the empty fields weren't visualised except after intentionally using "Add new .... "

File: app/models/service.rb [hunk 1]
```diff
@@ -308,6 +310,16 @@ def orderable?
...
+    def remove_empty_array_fields
+      multimedia.reject!(&:blank?) if multimedia.present?
+      use_cases_url.reject!(&:blank?) if use_cases_url.present?
+      certifications.reject!(&:blank?) if certifications.present?
+      standards.reject!(&:blank?) if standards.present?
+      open_source_technologies.reject!(&:blank?) if open_source_technologies.present?
+      changelog.reject!(&:blank?) if changelog.present?
+      related_platforms.reject!(&:blank?) if related_platforms.present?
+      grant_project_names.reject!(&:blank?) if grant_project_names.present?
+    end
```
Comment (wujuu) [inline app/models/service.rb:322]: Can't we refactor this into a for loop?

**Review by agpul — CHANGES_REQUESTED** [review]
For me, it works in a random way. Sometimes when I try to "Save" changes I need to remove one item a couple of times.

File: app/views/backoffice/services/_form.html.haml [hunk 2]
```diff
@@ -278,7 +278,7 @@
         %a.text-primary{ "data-action" => "click->service#addNewArrayField",
           "data-wrapper" => "service_related_platforms",
           "data-name" => "service[related_platforms[]]",
-          "data-class" => "form-control text optional" } Add additional entry
+          "data-class" => "form-control text optional" } Add new related platform
```
Comment (wujuu) [inline app/views/backoffice/services/_form.html.haml:281]: Shouldn't we use I18n translations?

**Review by wujuu — CHANGES_REQUESTED** [review]

**Comment by wujuu** [comment]
Also, shouldn't the behaviour to be to not show the 'Remove' button unless it's a second (and further) resource?
image

**Review by JanKapala — COMMENTED** [review]
Maybe some usage of locales would be nice

File: app/models/service.rb (hunk 1 above)
Comment (goreck888) [reply app/models/service.rb:320]: Done

**Comment by goreck888** [comment]
> Also, shouldn't the behaviour to be to not show the 'Remove' button unless it's a second (and further) resource?
> image

Honestly I think it's a bad idea. When service have eg. 3 entries and want to remove first then he couldn't do this. This solution is simple, working, and doesn't need unnecessary js code to handle this. And it fits to the logic in this form - look at PublicContacts, they're working in the same way.

**Review by agpul — CHANGES_REQUESTED** [review]
I still can remove only one element during one action.

**Review by wujuu — APPROVED** [review]

File: app/views/backoffice/services/_form.html.haml (hunk 2 above)
Comment (goreck888) [reply app/views/backoffice/services/_form.html.haml]: Done.

**Pull request merged**